app/scripts/main.js

```diff
@@ -7,6 +7,56 @@ window.LI = {
 
 (function (LI, $) {
   // Define private functions and variables here
+  var $search;
+
+  var onSearch = function(e) {
+    // Enter = 13
+    var searchTerm = $search.val();
+
+    if (e.keyCode === 13 && searchTerm) {
+      var search = {q: searchTerm};
+
+      $.when(
+        LI.global.fetchBooks(search),
+        LI.global.fetchAuthors(search),
+        LI.global.fetchPublishers(search),
+        LI.global.fetchCategories(search)
+      )
+        .then(function (booksResult, authorsResult, publishersResult, categoriesResult) {
+          var books = booksResult[0];
+          var authors = authorsResult[0].length ? authorsResult[0] : [{id: 0}];
+          var publishers = publishersResult[0].length ? publishersResult[0] : [{id: 0}];
+          var categories = categoriesResult[0].length ? categoriesResult[0] : [{id: 0}];
+
+          return $.when(
+            books,
+            LI.global.fetchAuthorBooks(authors[0].id, {}),
```

Comment:
We're just including the first result here. In the future we need to change this